describe-commit

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+#
+# Describe a commit in terms of refs
+#
+# Copyright 2017 Dan Nicholson <nicholson@endlessm.com>
+# Licensed under the new-BSD license (http://www.opensource.org/licenses/bsd-license.php)
+
+from __future__ import print_function
+
+from argparse import ArgumentParser
+from collections import defaultdict
+from gi import require_version
+require_version('OSTree', '1.0')
+from gi.repository import GLib, Gio, OSTree
+import sys
+
+def commit_describe(repo, checksum, all_refs=True):
+    refmap = defaultdict(list)
+
+    _, ref_info = repo.list_refs()
+    for ref, rev in ref_info.items():
+        # Walk the ref
+        depth = 0
+        while True:
+            _, commit = repo.load_variant_if_exists(OSTree.ObjectType.COMMIT,
+                                                    rev)
+            if commit is None:
```

Comment:
Actually this part is not coupled to the checksum that was passed in by the user. This loop is entirely about building up names for referenced commits and their parents. I can add a warning or error if the commit pointed to by the ref itself (`depth == 0`) does not exist. What do you think?